### PR TITLE
Update dataset API to use query params

### PR DESCRIPTION
**Pull request by marlosin**
Merged: ~~depends on https://github.com/jexia/jexia-sdk-js/pull/62~~

### CODE REVIEW

File: src/internal/executer.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class RequestExecuter {
     await this.systemInit;
     const token = await this.tokenManager.token(this.config.auth);
     return this.requestAdapter.execute(
-      this.getUrl(),
+      this.getURI(request.queryParams || []),
```
Comment (KutsenkoA) [inline src/internal/executer.ts]: I think you already set an empty array if there are no params in BaseQuery:
```
queryParams: this.query.compileToQueryParams() || [],
```
do you need to do it again here?

File: src/internal/executer.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class RequestExecuter {
...
       {
         headers: { Authorization: `Bearer ${token}` },
         body: request.body || {},
```
Comment (KutsenkoA) [inline src/internal/executer.ts]: Could `|| {}` be removed, what do you think?

File: src/internal/executer.ts (hunk 1 above)
Comment (marlosin) [reply src/internal/executer.ts]: I don't, I was just afraid of committing mistakes :laughing: 
If I just remove it typescript complains about the type `getURI` expects, so I removed ` || []` and changed method signature to accept empty array by default
```js
private getURI(queryParams: QueryParam[] = [])
```

File: src/internal/executer.ts (hunk 2 above)
Comment (marlosin) [reply src/internal/executer.ts]: done!

**Pull request merged**